File: src/main/java/com/ggardet/patient/spi/MedicalRecordService.java
package com.ggardet.patient.spi;

import com.ggardet.patient.spi.dto.MedicalRecordDTO;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public interface MedicalRecordService {
    MedicalRecordDTO create(final MedicalRecordDTO medicalRecordDTO);
    MedicalRecordDTO findById(final UUID id);
    List<MedicalRecordDTO> findAll();
    List<MedicalRecordDTO> findByPatientId(final UUID patientId);
    List<MedicalRecordDTO> findByDoctorId(final UUID doctorId);
    List<MedicalRecordDTO> findByDateRange(final LocalDateTime start, final LocalDateTime end);
    List<MedicalRecordDTO> findByFollowUpRequired(final Boolean followUpRequired);
    MedicalRecordDTO update(final UUID id, final MedicalRecordDTO medicalRecordDTO);
    void deleteById(final UUID id);
}

File: src/main/kotlin/com/ggardet/patient/spi/MedicalRecordService.kt
package com.ggardet.patient.spi

import com.ggardet.patient.spi.dto.MedicalRecordDTO
import java.time.LocalDateTime
import java.util.*

interface MedicalRecordService {
    fun create(medicalRecordDTO: MedicalRecordDTO): MedicalRecordDTO
    fun findById(id: UUID): MedicalRecordDTO
    fun findAll(): List<MedicalRecordDTO>
    fun findByPatientId(patientId: UUID): List<MedicalRecordDTO>
    fun findByDoctorId(doctorId: UUID): List<MedicalRecordDTO>
    fun findByDateRange(start: LocalDateTime, end: LocalDateTime): List<MedicalRecordDTO>
    fun findByFollowUpRequired(followUpRequired: Boolean): List<MedicalRecordDTO>
    fun update(id: UUID, medicalRecordDTO: MedicalRecordDTO): MedicalRecordDTO
    fun deleteById(id: UUID)
}